lib/pbench/agent/tool_data_sink.py

```diff
@@ -614,16 +818,15 @@ def _wait_for_all_data(self):
 
     def _change_tm_tracking(self, curr, new):
         """_change_tm_tracking - if we have a tool meister tracking dictionary
-        update the posted state from the current expected value to the target
+        update the posted action from the current expected value to the target
         new value.
 
         No changes take place if the tool meister tracking dictionary does not
         exist yet.
 
         Assumes self._lock is already acquired by our caller.
         """
-        if self._tm_tracking is None:
-            return
+        assert self._tm_tracking is not None, "Logic bomb!  self._tm_tracking is None"
```

Comment:
Doesn't this conflict with the "No changes take place if the tool meister tracking dictionary does not exist yet" comment above??